sql/analyzer/indexed_joins.go

```diff
@@ -402,6 +422,10 @@ type joinIndex struct {
 	table string
 	// The index that can be used in this join, if any. nil otherwise
 	index sql.Index
+	// If this is set, the join condition is a top-level OR
+	// expression which can potentially make use of two (or more)
```

Comment:
A bit confusing... this field stores exactly two joinIndexes, which themselves may be OR conditions